src/Serializer/AbstractItemNormalizer.php

```diff
@@ -116,7 +116,7 @@ public function hasCacheableSupportsMethod(): bool
      */
     public function normalize($object, $format = null, array $context = [])
     {
-        if ($this->handleNonResource && ($context['api_normalize'] ?? false) || null !== $outputClass = $this->getOutputClass($this->getObjectClass($object), $context)) {
+        if ($this->handleNonResource) {
```

Comment:
```php
if ($this->handleNonResource && ($context['api_normalize'] ?? false)) {
```

Reply:
Or perhaps we should do the check differently:

```php
if ($this->handleNonResource) {
    if (!($context['api_normalize'] ?? false)) {
        throw new UnexpectedValueException('"api_normalize" must be set to true in context to normalize non-resource');
    }

    return parent::normalize($object, $format, $context);
}
```

Reply:
I think the call to `initContext` is useless.

Reply:
no we need to change the resource_class

Reply:
Uhh, because `AbstractObjectNormalizer::normalize` will call `getAttributeValue`. Gotcha!

Reply:
Hmm, I don't see where it'd be used... We're going to the parent and not coming back.